src/videojs-media-sources.js

```diff
@@ -301,9 +365,9 @@
           offset += segment.byteLength;
         });
 
-        // Set timestampOffset if we have been given one
-        if (this.timestampOffset !== undefined) {
-          destinationBuffer.timestampOffset = this.timestampOffset;
+        // Set timestampOffset if it was changed before this append
```

Comment:
Maybe.

The benefit to doing it this way is that we don't have to worry about the buffers existing when `timestampOffset` is set - the offset is set only when we have done everything we needed to do to create the sourcebuffers but right before the actual `append`.

The real source buffers can be created long after the "virtual" source buffer is created. You may want to create a Media-Source object and set the `timestampOffset` before you ever start pumping data into the transmuxer (and therefore create the source buffers). This allows for that contingency without introducing too many complications.